airflow/providers/amazon/aws/hooks/glue.py

```diff
@@ -0,0 +1,214 @@
+            else:
+                self.log.info("Polling for AWS Glue Job %s current run state with status %s"
+                    , job_name, job_run_state)
+                time.sleep(6)
+
+    def get_or_create_glue_job(self):
+        """
+        Creates(or just returns) and returns the Job name
+        :return:Name of the Job
+        """
+        glue_client = self.get_conn()
+        try:
+            get_job_response = glue_client.get_job(JobName=self.job_name)
+            self.log.info("Job Already exist. Returning Name of the job")
+            return get_job_response['Job']['Name']        
+        except glue_client.exceptions.EntityNotFoundException:
+            try:
+                self.log.info("Job Not exist. Now creating and running AWS Glue Job")
+                s3_log_path = "s3://{bucket_name}/{logs_path}{job_name}"\
+                    .format(bucket_name=self.s3_bucket,
+                            logs_path=self.S3_GLUE_LOGS,
+                            job_name=self.job_name)
+
+                execution_role = self.get_iam_execution_role()
+                script_location = self._check_script_location()
+                create_job_response = glue_client.create_job(
+                    Name=self.job_name,
+                    Description=self.desc,
+                    LogUri=s3_log_path,
+                    Role=execution_role['Role']['RoleName'],
+                    ExecutionProperty={"MaxConcurrentRuns": self.concurrent_run_limit},
+                    Command={"Name": "glueetl", "ScriptLocation": script_location},
+                    MaxRetries=self.retry_limit,
+                    AllocatedCapacity=self.num_of_dpus
+                )
+                return create_job_response['Name']   
+            except Exception as general_error:
+                raise AirflowException(
+                    'Failed to create aws glue job, error: {error}'.format(
+                        error=str(general_error)
+                    )
+                )    
+
+    def _check_script_location(self):
+        """
+        :return: S3 Script location path
+        """
+        if self.script_location[:5] == self.S3_PROTOCOL:
+            return self.script_location
+        elif os.path.isfile(self.script_location):
+            s3 = self.get_resource_type('s3', self.region_name)
+            script_name = os.path.basename(self.script_location)
+            s3.meta.client.upload_file(self.script_location,
+                                       self.s3_bucket,
+                                       self.S3_ARTIFACTS_PREFIX + script_name)
+
+            s3_script_path = "s3://{s3_bucket}/{prefix}{job_name}/{script_name}" \
+                .format(s3_bucket=self.s3_bucket,
+                        prefix=self.S3_ARTIFACTS_PREFIX,
+                        job_name=self.job_name,
+                        script_name=script_name)
+            return s3_script_path
+        else:
+            return None
```

Comment:
Can you check if you can reuse some code of `S3Hook` ? For example there is a `upload_file` in it.

I am feeling not so comfortable mixing so many aws services on this detailed implementation level. We have special hooks per AWS service which you can use to interact with them.

Reply:
Hello, I have fixed other changes but for this one I am not sure if I will be able to use upload file of s3 hook as I havent used it and I quite dont get mapping of prefix to key in the conversion.
Is it possible to go without this or if someone else can convert it?

Reply:
It is basically the same.
And you also don't need the `if`, `elif`, `else` here.
I would just go for sth. like:
```python
s3_hook = S3Hook(aws_conn_id=self.aws_conn_id)
s3_hook.load_file(self.script_location, self.s3_bucket, self.S3_ARTIFACTS_PREFIX + script_name)
return f's3://{self.s3_bucket}/{self.S3_ARTIFACTS_PREFIX}{job_name}/{script_name}'
```

But I am wondering about the formatted string you are returning. You are loading the file to `f's3://{self.s3_bucket}/{self.S3_ARTIFACTS_PREFIX + script_name}'`. Can you explain your formatted string, please?

Reply:
@feluelle Thanks for the response and snippet. 
This piece of code was written by @OElesin, Yes i think `{job_name}` should not be there as it is not in the upload path.

Reply:
Actually I think it would be better to expect the `self.script_location` to be a valid path to s3 from the beginning. If you need to upload some files first do this in the operator by calling `S3Hook.load_file` first.

Reply:
Hi @feluelle , Just committed the changes, would be very thankful if you can have a look on changes, if they are fine, i will mark this resolved.